src/includes/dinit-iostream.h

```diff
@@ -0,0 +1,658 @@
+    }
+
+    istream(const int tfd, std::unique_ptr<streambuf> passed_buf) noexcept
+    {
+        fd = tfd;
+        buf = std::move(passed_buf);
+    }
+
+    istream(const istream &) = delete;
+
+    void operator=(const istream &) = delete;
+
+    istream(const istream &&) = delete;
+
+    void operator=(const istream &&) = delete;
+
+    // Open file path which specified in path or as open(2) parameter with flags or modes.
+    //
+    // Note: O_RDONLY is hard-coded in internal call to POSIX open(2).
+    //
+    // Note: Opening an already opened stream is not allowed and caller must close the stream
+    // before open it again.
+    //
+    // Returns: true on success, false on failure.
+    //
+    // Non _nx variant may throw:
+    // dio::iostream_system_err on POSIX open(2) failure.
+    bool open_nx(const char *path) noexcept;
+    bool open(const char *path);
+
+    bool open_nx(const char *path, const int flags) noexcept;
+    bool open(const char *path, const int flags);
+
+    bool open_nx(const char *path, const int flags, const mode_t mode) noexcept;
+    bool open(const char *path, const int flags, const mode_t mode);
+
+    // Close the current file descriptor.
+    //
+    // Returns: true on success, false on failure cases.
+    //
+    // Non _nx variant may throw:
+    // dio::iostream_system_err on system I/O failure.
+    bool close_nx() noexcept;
+    bool close();
+
+    // Returns: current bits in stream based on io_states enum.
+    int current_state() noexcept;
+
+    // Returns: true when there is no set bits/vars otherwise false.
+    bool good() noexcept;
+
+    // Returns: true or false based on eofbit is set or not.
+    bool eof() noexcept;
+
+    // Returns: true or false based on buffer_failbit is set or not.
+    bool buffer_failure() noexcept;
+
+    // Returns: true or false based on string_failbit is set or not.
+    bool string_failure() noexcept;
+
+    // Returns: 0 or errno based on io_error value.
+    int io_failure() noexcept;
+
+    // Resets failure bits/vars to false (or 0 about io_error).
+    // Note: buffer_failbit will not be reset.
+    void clear() noexcept;
+
+    // Reads and returns one character from file descriptor.
+    //
+    // Returns: Read character and success of operation as getc_result object.
+    //
+    // Non _nx variant may throw:
+    // dio::iostream_eof on EOF.
+    // dio::iostream_internal_err on Buffer failure (buffer did not allocated at construction time).
+    // dio::iostream_system_err on system I/O failure.
+    getc_result getc_nx() noexcept;
+    getc_result getc();
+
+    // Reads and stores one line from fd into given std::string.
+    //
+    // istream provides the get_line() function to read from its file descriptor until the delimiter
+    // character is found (usually '\n') and write it to the given std::string. Also, the given
+    // std::string will be erased when there is something in the buffer on each call.
```

Comment:
"Also, the given std::string will be erased when there is something in the buffer on each call."
->
"The previous contents of the string are lost."

Reply:
(or "The previous contents of the string are erased.")